Hero/Hero.swift

```diff
@@ -86,6 +86,20 @@ internal extension Hero {
   }
 }
 
+fileprivate extension Hero {
+  func closureProcessForHeroDelegate<T: UIViewController>(vc: T, closure: (HeroViewControllerDelegate)->()) {
+    switch vc {
+      case let fvc as HeroViewControllerDelegate:
+        closure(fvc)
+      case let fvc as UINavigationController:
```

Comment:
Realized a bug. If vc is a UINavigationController, and if both the vc and its topViewController implements HeroViewControllerDelegate. The previous implementation will first call to vc then to its topViewController.
Here it is only